packages/react-spinbutton/Spec.md

```diff
@@ -2,62 +2,381 @@
+
+Native number inputs are meant strictly for number input but what constitutes number input is inconsistent across browsers ([see this Bugzilla issue for details](https://bugzilla.mozilla.org/show_bug.cgi?id=1398528)). You can easily see this on [MDN's simple example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#a_simple_number_input). In Edge 96 you can enter exponential numbers like "1e+343434" but not arbitrary strings like "cats". On the same example in Firefox 95 you can enter both "1e+343434" and "cats".
+
+Inspecting a native number input with devtools shows that it implements the [spinbutton ARIA attributes as described by WAI-ARIA](https://www.w3.org/TR/wai-aria-practices/#spinbutton)
+
+#### Examples
+
+- [Controlling Step Size](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#controlling_step_size)
+- [Specifying Minimum and Maximum Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#specifying_minimum_and_maximum_values)
+- [Allowing Decimal Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#allowing_decimal_values)
+- [Offering Suggested Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#offering_suggested_values)
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+### Uncontrolled Example
+
+```tsx
+<SpinButton label="Uncontrolled Example" defaultValue="1" />
+```
+
+### Controlled Example
+
+```tsx
+type SpinButtonChangeData = {
+  value: number; // The "raw value", e.g., 2
+  formattedValue: string; // The "formatted value", e.g., "$2.00"
+  prevValue: number;
+  prevFormattedValue: string;
+};
+
+const [value, setValue] = useState<number>(2);
+const onControlledExampleChange = (_event, data: SpinButtonChangeData) => {
+  setValue(data.value);
+};
+
+<SpinButton label="Controlled Example" value={value} onChange={onControlledExampleChange} />;
+```
+
+### Custom Display Format Example
+
+```tsx
+type SpinButtonChangeData = {
+  value: number;
+  formattedValue: string;
+  prevValue: number;
+  prevFormattedValue: string;
+};
+
+type SpinButtonFormatter = (value: number) => string;
+type SpinButtonParser = (formattedValue: string) => number;
+
+const [value, setValue] = useState<number>(3);
+const onControlledExampleChange = (_event, data: SpinButtonChangeData) => {
+  setValue(data.value);
+};
+
+// Takes a number in and returns a formatted string
+// Ex: 12 becomes "12 pt"
+const fontFormatter: SpinButtonFormatter = value => {
+  return `${value} pt`;
+};
+
+// Takes a formatted string in and returns a number
+// Ex: "12 pt" becomes 12
+const fontParser: SpinButtonParser = formattedValue => {
+  return parseFloat(formattedValue);
+};
```

Comment:
Is it necessary to have the parser ? It would make the component much simpler to only accept and manage numerical values. I can't see a scenario where users would need to pass a formatted value into the component.

Users should parse their own numerical values for the component if they need to

Reply:
I see, this is necessary for formatting since the input is actually a text input

Reply:
should we enforce setting a custom parser if a custom formatter is used ?

Reply:
> should we enforce setting a custom parser if a custom formatter is used ?

I think this is a good thing to do.

Reply:
> Is it necessary to have the parser ? It would make the component much simpler to only accept and manage numerical values. I can't see a scenario where users would need to pass a formatted value into the component.

I agree that it adds a fair bit of complexity and the implementation of this component will be much simpler if we only accept numeric input. I'm following up on the need for this with our partners.